mlir/lib/Dialect/Tensor/Transforms/FoldTensorSubsetOps.cpp

```diff
@@ -154,6 +163,18 @@ LogicalResult InsertSliceOfTransferWriteOpFolder::matchAndRewrite(
   return success();
 }
 
+bool InsertSliceOfTransferWriteOpFolder::doesTransferWriteCoverInsertSlice(
+    vector::TransferWriteOp writeOp) {
+  if (writeOp.getVectorType().hasStaticShape() &&
```

Comment:
I think the vector type can only have static shapes, so we don't need this check?